pkg/materialize/source_table_kafka.go

```diff
@@ -0,0 +1,389 @@
+package materialize
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/jmoiron/sqlx"
+)
+
+type SourceTableKafkaParams struct {
+	SourceTableParams
+}
+
+var sourceTableKafkaQuery = `
+	SELECT
+		mz_tables.id,
+		mz_tables.name,
+		mz_schemas.name AS schema_name,
+		mz_databases.name AS database_name,
+		mz_sources.name AS source_name,
+		source_schemas.name AS source_schema_name,
+		source_databases.name AS source_database_name,
+		mz_kafka_source_tables.topic AS upstream_table_name,
```

Comment:
I have a PR open now to add the `envelope`, `key_format` and `value_format` columns to this table https://github.com/MaterializeInc/materialize/pull/30076

Reply:
Neat! I created a tracking issue for this for the moment: https://github.com/MaterializeInc/terraform-provider-materialize/issues/665

Should be ok to handle in a follow up PR later on!